test/functional/geckodriver-e2e-specs.js

```diff
@@ -0,0 +1,91 @@
+// transpile:mocha
+
+import Geckodriver from '../../lib/geckodriver';
+import chai from 'chai';
+import chaiAsPromised from 'chai-as-promised';
+
+let should = chai.should();
+chai.use(chaiAsPromised);
+
+const caps = {
+  capabilities: {
+    alwaysMatch: {
+      browserName: 'firefox',
+      'moz:firefoxOptions': {
```

Comment:
this what I was talking about - there should be a way to provide these options from Appium caps

Reply:
how would the final schema would look like?

Reply:
we could have a capability: `firefoxOptions` of type Object.
In this capability it should be possible to set all these options you've set here and they would be used as defaults while building the actual driver capabilities unless overridden by high-level specific caps (see https://lodash.com/docs/4.17.15#defaultsDeep)